README.md

```diff
@@ -1,8 +1,119 @@
 # SQLair
 
-Friendly type mapping for SQL databases.
+SQLair is a package for Go that acts as a compatibility layer between Go and SQL databases. It allows for easy mapping between Go objects and query arguments and results.
+
+SQLair allows you to write SQL with SQLair input/output expressions included where mapping is needed. These expressions indicate the Go objects to map the database arguments/results into. SQLair is not an ORM and will leave all parts of the query outside of the expressions untouched.
+
+The API can be found at [pkg.go.dev](https://pkg.go.dev/github.com/canonical/sqlair).
+A demo can be found at [demo/demo.go](demo/demo.go).
+
+There will also soon be a full tutorial but that is currently a work in progress.
+### Motivation
+When writing an SQL query with `database/sql` in Go there are multiple points of redundency/failure:
+
+- The order of the columns in the Query must match the order of columns in `Scan`
+- The columns from the Query must be manuelly match to their destinations
+- If the columns needed change all queries must be changed
+
+For example, when selecting a particular `Person` from a database, instead of the query: 
+```SQL
+SELECT name_col, id_col, gender_col FROM person WHERE manager_col = ?
+```
+In SQLair you could write:
+```SQL
+SELECT &Person.* FROM person WHERE manager_col = $Manager.name
+```
+This results from this second query could then be directly decoded in the the `Person` struct.
+
+
+# Usage
+The first step when using SQLair is to tag your structs. The "`db`" tag is used to map between the database column names and struct fields.
+
+For example:
+```Go
+type Person struct {
+	Name	string	`db:"name_col"`
+	ID 	int64 	`db:"id_col"`
+	Gender	string  `db:"gender_col"`
+}
+```
+It is important to note that SQLair __needs__ the fields to be public in order read from them and write to them.
+
+To run a query with SQLair you need to create your `sqlair.DB` object. This is done by wrapping a `sql.DB` with:
+```Go
+db := sqlair.NewDB(sqldb)
+```
+
+Now to get a `Person` from the database.
+
+```Go
+stmt := sqlair.MustPrepare(
+    "SELECT &Person.* FROM people",
+    Person{},
+)
+
+var person := Person{}
+err := db.Query(stmt).One(&person)
+```
+
+Or maybe we want all the people from the database
+
+```Go
+var people := []Person{}
+err := db.Query(stmt).All(&people)
+```
+
+Or maybe some more precise control is needed
```

Comment:
needed, where we would like to iterate over query results and do something with each result, like so:

Reply:
Added.